### DISCUSSION TITLE
OSGT output incorrect alignement

### DISCUSSION THREAD

**Discussion by llagardeOSE** [openscenegraph/OpenSceneGraph#1321] [Q&A]
Hi, 
I recently recompiled OSG using visual studio 2022.

Writing OSGT files now gives:
```
    osg::Geometry {
      UniqueID 11 
      Name "group" 
StateSet TRUE {
osg::StateSet {
          UniqueID 12 
          AttributeList 2 {
osg::Material {
              UniqueID 13 
              Ambient TRUE Front 0.361307 0.361307 0.361307 1 Back 0.361307 0.361307 0.361307 1 
Diffuse TRUE Front 0.361307 0.361307 0.361307 1 Back 0.361307 0.361307 0.361307 1 
Specular TRUE Front 0 0 0 1 Back 0 0 0 1 
Emission TRUE Front 0 0 0 1 Back 0 0 0 1 
Shininess TRUE Front 0 Back 0 
}
```

The `OutputIterator::isEndl` method does not always recognize the `std::endl` modifier and the next written line is not indented.
This behavior is consistent, the same lines are always indented the same way.

Any idea ?

**Comment by robertosfield** [comment]
I'm not a Windows expert so will have to defer to others to see if they have insights what VisualStudio 2022 might be doing differently.  Which version of VisualStudio did this last work as expected?

Does the resulting .osgt file still read back in?

What version of OSG are you testing against?

**Reply by llagardeOSE** [reply]
The previous compilation of OSG was with visual 2010. The OSGT files were good.
The files are still readable, it's only "human analysis" that is complicated.

**Reply by robertosfield** [reply]
I'm not in a position to go debugging on the OSG right now so will have to leave this to others to investigate and resolve.  It's good to hear at least it's not a critical error.

**Comment by AnyOldName3** [comment]
It's been the best part of a year (I thought I'd been checking the OSG GitHub discussions more often than it seems I really have) but I'd suspect that this is related to something we've had to fix in the OpenMW OSG fork: https://github.com/OpenMW/osg/pull/24

**Reply by llagardeOSE** [reply]
Thanks
